### DISCUSSION TITLE
Is there a micropython library for face recognition? It is also ok to make C into mpy file.

### DISCUSSION THREAD

**Discussion by Vincent1-python** [micropython/micropython#16287] [ESP32]
Is there a micropython library for face recognition? It is also ok to make C into mpy file.

**Comment by bidrohini** [comment]
I could only find this video: https://www.youtube.com/watch?v=TA0dJf2njFQ

**Comment by andrewleech** [comment]
The OpenMV project has features like this, though that's targeting their own hardware as opposed to any generic micropython board. Do you have any particular hardware you're trying to use?

**Comment by Vincent1-python** [comment]
esp32s3



At 2024-11-23 17:30:52, "Andrew Leech" ***@***.***> wrote:

The OpenMV project has features like this, though that's targeting their own hardware as opposed to any generic micropython board. Do you have any particular hardware you're trying to use?

**Comment by mattytrentini** [comment]
Depending on your requirement, an alternative way would be to use a [Person Sensor](https://github.com/usefulsensors/person_sensor_docs). Cheap and very easy to integrate, though inappropriate if high reliability or security is a concern.

**Comment by jonnor** [comment]
emlearn-micropython has support for Convolutional Neural Networks, which is the most commonly used machine learning model for face recognition. You will have to train the model yourself, however. This example illustrates the usage https://github.com/emlearn/emlearn-micropython/tree/master/examples/mnist_cnn
